torch/_subclasses/fake_impls.py

```diff
@@ -455,9 +457,28 @@ def masked_select(fake_mode, func, self, mask):
         has_free_symbols,
     )
 
+    # If num elements is expressed symbolically, calculate
+    # the concrete value based on upper bounds. Otherwise,
+    # we can set max val directly.
     if not has_free_symbols(self.numel()):
-        if self.numel() > 2:
-            maxval = int(self.numel())
+        num_elements = int(self.numel())
+    else:
+        if len(self.shape) != 0:
+            num_elements = 1
+            for size in self.shape:
+                if isinstance(size, int):
+                    num_elements *= size
+                elif isinstance(size, torch.SymInt):
+                    sym_node = size.node
+                    sym_range = bound_sympy(
+                        sym_node.expr, sym_node.shape_env.var_to_range
+                    )
+                    if isinstance(sym_range.upper, IntInfinity):
+                        num_elements = sys.maxsize - 1
+                        break
+                    num_elements *= int(sym_range.upper)
```

Comment:
I think you do not have to do it this manually. Instead, do math.prod(self.shape) and then read out the var_to_range upper bound from the result. You can also generate a minimum bound as well this way.

Reply:
Hmm what should be the code here? I'm getting the following output on the example in my unit test, not sure what to do with these.

```
>> math.prod(self.shape)
5*s0*s1
>> math.prod(self.shape).node.shape_env.var_to_range
{s0: VR[2, 100], s1: VR[2, 7], u0: VR[-int_oo, int_oo]}
```

Reply:
You still have to use bound_sympy